connector/topologyconnector/internal/mapping_handler.go

```diff
@@ -114,6 +118,20 @@ func (v *MappingContext[T]) ExecuteMapping(
 	ctx context.Context,
 	evalCtx *ExpressionEvalContext,
 ) {
+	// Only perform pre-mapping deduplication when we have selective expression references.
+	// If a mapping only depends on resource/scope (which are always hashed unconditionally),
+	// no ExpressionRefSummary is produced and deduplication is intentionally skipped.
+	// This avoids over-deduplication when no additional selectivity can be proven from
+	// the mapping expressions.
+	expressionRef := v.BaseCtx.ExpressionRefSummaries[v.Mapping.GetIdentifier()]
```

Comment:
Can we hide the `ExpressionRefSummaries` here and have the `Deduplicator` manage them? `ShouldSend` already has all the arguments it needs to resolve them, so it seems unnecessary that we have to provide this separately.

Reply:
never mind. I see you already managed to implement it.